api/gateway/basecommerce/basecommerce.cfc

```
@@ -0,0 +1,281 @@
+			}
+		}
+		
+		return local.response;
+	}
+
+	private any function createResponse() {
+		return createObject("component", "cfpayment.api.gateway.basecommerce.response").init(argumentCollection = arguments, service = getService());
+	}
+	
+	private any function CreateEmptyResponse(struct payload=structNew()) {
+		local.response = structNew();
+		
+		local.response.message = '';
+		local.response.result = '';
+		local.response.status = 0;
+		local.response.statusCode = '';
+		local.response.testMode = true;
+		local.response.requestData = structNew();
+		local.response.requestData['gateway_url'] = '';
+		local.response.requestData['headers'] = structNew();
+		local.response.requestData['http_method'] = '';
+		local.response.requestData['payload'] = arguments.payload;
+		
+		return local.response;
+	}
+
+	//Implement primary methods
+	public any function purchase(required any money, any account, struct options=structNew()) {
+		local.argumentCollection = structNew();
+		local.argumentCollection.money = arguments.money;
+		if(isDefined('arguments.account')) local.argumentCollection.money = arguments.account;
+		local.argumentCollection.options = arguments.options;
+		local.argumentCollection.options.batType = 'XS_BAT_TYPE_CREDIT';
+		return transaction(argumentCollection = local.argumentCollection);
+	}
+	
+	public any function credit(required any money, any account, struct options=structNew()) {
+		local.argumentCollection = structNew();
+		local.argumentCollection.money = arguments.money;
+		if(isDefined('arguments.account')) local.argumentCollection.money = arguments.account;
+		local.argumentCollection.options = arguments.options;
+		local.argumentCollection.options.batType = 'XS_BAT_TYPE_DEBIT';
+		return transaction(argumentCollection = local.argumentCollection);
+	}
+	
+	public any function store(required any account, struct options=structNew()) {
+		switch(getService().getAccountType(arguments.account)) {
+			case 'creditcard':
```

probably better to throw an error if it's anything other than an eft object as unsupported.  Check the gateways for other examples of unsupported types.